### ISSUE TITLE
Settings access fails

### ISSUE THREAD

**Issue by mvl22** [cyclestreets/android#456]
User bug report:

> I have been using CycleStreets on my phone for 4 years and love the offline maps. I changed from a Sony to a Google Pixel 4a recently and now I cannot access the settings. When I click on the settings option in the menu it jumps to my phone's home screen. A Samsung S10 phone, to which a family member can provide temporary access for me, has no issues though.

**Comment by oliverlockwood** [comment]
Thanks @mvl22.

I can see in the Play console a number of recent instances of the following stack trace:

```
java.lang.NullPointerException: 
  at net.cyclestreets.util.MapPack.availableMapPacks (MapPack.java:38)
  at net.cyclestreets.SettingsFragment.populateMapFileList (SettingsFragment.kt:137)
  at net.cyclestreets.SettingsFragment.setupMapFileList (SettingsFragment.kt:132)
  at net.cyclestreets.SettingsFragment.onCreate (SettingsFragment.kt:53)
  at androidx.fragment.app.Fragment.performCreate (Fragment.java:2684)
  at androidx.fragment.app.FragmentStateManager.create (FragmentStateManager.java:280)
  at androidx.fragment.app.FragmentManager.moveToState (FragmentManager.java:1175)
  at androidx.fragment.app.FragmentTransition.addToFirstInLastOut (FragmentTransition.java:1255)
  at androidx.fragment.app.FragmentTransition.calculateFragments (FragmentTransition.java:1138)
  at androidx.fragment.app.FragmentTransition.startTransitions (FragmentTransition.java:136)
  at androidx.fragment.app.FragmentManager.executeOpsTogether (FragmentManager.java:1989)
  at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute (FragmentManager.java:1947)
  at androidx.fragment.app.FragmentManager.execPendingActions (FragmentManager.java:1849)
  at androidx.fragment.app.FragmentManager$4.run (FragmentManager.java:413)
  at android.os.Handler.handleCallback (Handler.java:938)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loop (Looper.java:223)
  at android.app.ActivityThread.main (ActivityThread.java:7660)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:592)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:947)
```

The Android IDE even warns of a possible NPE in this code path, so I'll put some defensive coding in place shortly.

Perhaps you can advise them that we are aiming to get an update out within the next month, that will fix this NPE.
However they should be aware that offline maps are unlikely to be working in such a release (unless @jezhiggins suddenly emerges from hibernation and finishes his changes in that area.)

**Comment by oliverlockwood** [comment]
Closed by #457
